colossus-metrics/src/main/scala/colossus/metrics/collectors/Rate.scala

```diff
@@ -24,15 +26,65 @@ class Rate private[metrics](val address: MetricAddress, pruneEmpty: Boolean)(imp
       Map(address -> snap, address / "count" -> totals.snapshot(pruneEmpty, false))
     }
   }
-    
-
 }
 
-object Rate {
+object Rate extends CollectorConfigLoader {
+
+  private val DefaultConfigPath = "colossus.metrics.collectors-defaults.rate"
+
+  /**
+    * Create a Rate with the following address.  This will use the "colossus.metrics" config path to locate configuration.
+    * @param address The MetricAddress of this Rate.  Note, this will be relative to the containing MetricSystem's metricAddress.
+    * @param collection The collection which will contain this Collector.
+    * @return
+    */
+  def apply(address : MetricAddress)(implicit collection : Collection) : Rate = {
+    apply(address, MetricSystem.ConfigRoot)
+  }
+  /**
+    * Create a Rate with the following address.  Note, the address will be prefixed with the MetricSystem's root.
+    * Configuration is resolved and overlaid as follows('metricSystemConfigPath' is the configPath parameter, if any, that was
+    * passed into the MetricSystem.apply function):
+    * 1) configPath.address
+    * 2) metricSystemConfigPath.collectors.defaults.rate
+    * 3) colossus.metrics.collectors.defaults.rate
+ *
+    * @param address The MetricAddress of this Rate.  Note, this will be relative to the containing MetricSystem's metricAddress.
+    * @param configPath The path in the ConfigFile that this rate is located.
+    * @param collection The collection which will contain this Collector.
+    * @return
+    */
+  def apply(address : MetricAddress, configPath : String)(implicit collection : Collection) : Rate = {
+
+    val params = resolveConfig(collection.config.config, s"$configPath.$address", DefaultConfigPath)
+    apply(address, params.getBoolean("prune-empty"))
+  }
 
+  /**
+    * Create a new Rate which will be contained by the specified Collection
+    * @param address The MetricAddress of this Rate.  Note, this will be relative to the containing MetricSystem's metricAddress.
+    * @param pruneEmpty Instruct the collector to not report any values for tag combinations which were previously empty.
+    * @param collection The collection which will contain this Collector.
+    * @return
+    */
   def apply(address: MetricAddress, pruneEmpty: Boolean = false)(implicit collection: Collection): Rate = {
     collection.getOrAdd(new Rate(address, pruneEmpty))
   }
 }
 
+private[metrics] trait CollectorConfigLoader {
```

Comment:
This trait should probably get moved into its own file, since other things besides rate mix it in.